SH4RE/SH4RE/Messages/MessagesChat.swift

```diff
@@ -0,0 +1,283 @@
+//
+//  MessagesChat.swift
+//  SH4RE
+//
+//  Created by Hannah Brooks on 2023-01-17.
+//
+
+import SwiftUI
+import Firebase
+
+class ChatLogViewModel: ObservableObject {
+    
+    @Published var chatText = ""
+    @Published var errorMessage = ""
+    
+    @Published var chatMessages = [ChatMessage]()
+    
+    var chatUser: ChatUser?
+    
+    init(chatUser: ChatUser?) {
+        self.chatUser = chatUser
+        
+        fetchMessages()
+    }
+    
+    var firestoreListener: ListenerRegistration?
+    
+    func fetchMessages() {
+        guard let fromId = FirebaseManager.shared.auth.currentUser?.uid else { return }
+        guard let toId = chatUser?.uid else { return }
+        firestoreListener?.remove()
+        chatMessages.removeAll()
+        firestoreListener = FirebaseManager.shared.firestore
+            .collection(FirebaseConstants.messages)
+            .document(fromId)
+            .collection(toId)
+            .order(by: FirebaseConstants.timestamp)
+            .addSnapshotListener { querySnapshot, error in
+                if let error = error {
+                    self.errorMessage = "Failed to listen for messages: \(error)"
```

Comment:
i am assuming this isnt shown to the user, but if it is, we should make a ticket to change error messages to only the debugging env

Reply:
@bbryanb333

Reply:
It's not show to user